service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -379,7 +382,8 @@ public void send(String deviceId, String moduleId, Message message) throws IOExc
             {
                 future.completeExceptionally(e);
             }
-        });
+        }).run();
```

Comment:
I'm a bit confused as to what we want done here - do we want to call open() on a different thread rather than the current thread? If so, why? And also, if we want to run it on a different thread don't we need to call start()?

Reply:
>do we want to call open() on a different thread rather than the current thread?

 Yes, because this method is ```openAsync()```

> If so, why?

If we run ```open()``` on the current thread, then it isn't async at all

> And also, if we want to run it on a different thread don't we need to call start()?

You are right, I shouldn't be using ```run()``` here. I should be using ```start()```

Reply:
This mess of async flavors of APIs is making me consider just removing them from our v2 client. I'll bring this up with our team in standup soon. They don't really add anything that a customer couldn't create a wrapper to handle